docs/DeveloperGuide.md

```diff
@@ -2,12 +2,111 @@
+4. Users can choose to search through the expense list or income list.
+5. If users searches through the income list, either the description or date will be used to search through the income list.
+6. If users searches through the expense list, either the category or date will be used to search through the expense list.
+7. The execute method will then be called to search through the list and return the list of expenses or incomes that matches the search criteria.
+
+![FindCommand_SequenceDiagram.png](https://github.com/AY2324S1-CS2113-T18-3/tp/blob/master/images/FindCommand_SequenceDiagram.png?raw=true)
+
+### GetFromTxt Command
+The GetFromTxt Command is facilitated by `IOException`,`Scanner`, `KaChinnnnngException`, `IncomeList` and `ExpenseList`
+1. The user start the program and the command will start automatically with a default path
+2. Once the command executed, it will try to create an txt file on the default path
+3. If file failed to create, throw an `IOException`
+4. Catch the `IOExceotion` and print error messages
+5. Create a `Scanner` Object s to get content from txt file
+6. If amount in txt file exceed limit or incorrect, throw `KaChinnnningException`
+7. Read the content on the file and add corresponding object to incomes or expenses list
+
+![GetFromTxtCommands_SequenceDiagram.png](https://github.com/AY2324S1-CS2113-T18-3/tp/blob/master/images/GetFromTxt.png?raw=true)
+
+### ExchangeRateManager
+The Sequence Diagram below shows how the components interact with each other for the scenario 
+where the user issues the command `update exchange rate USD 0.8`.
+
+![UpdateExchangeRate_SequenceDiagram.png](https://github.com/AY2324S1-CS2113-T18-3/tp/blob/master/images/UpdateExchangeRate_SequenceDiagram.png?raw=true)
+
+This is facilitated by `Duke`, `UpdateExchangeRateCommand`, `ExchangeRateManager`'and `ExchangeRateFileHandler`.
+
+### SaveToTxt Command
+The SaveToTxt Command is facilitated by `FileWriter` ,`IncomeList` and `ExpenseList`
+1. The user start the program and the function will run everytime after user input a command
+2. Once the command executed, the function saveIncomeAndExpense will run automatically
+3. It will run the function saveIncomeToTextFile first, the saveIncomeToTextFile will create a FileWriter and loop through the incomes list
+4. It will get description, date, and amount from the income list for every single income record in the list and save the information into the text file
+5. After that, the saveIncomeToTextFile will then call the saveExpenseToTextFile function 
+6. The saveExpenseToTextFile will create a new FileWriter and loop through the expenses list 
+7. It will get description, date, amount and type from the expenses list for every single expense record in the list and save the information into the text file
+
+![SaveToTxt_SequenceDiagram.png](https://github.com/AY2324S1-CS2113-T18-3/tp/blob/master/images/SaveToTxt.png?raw=true)
```

Comment:
In the diagram, it shows the fwe:FileWriter object being destroyed at the end of the program, but it gets instantiated again underneath the X? Is this how the program is intended to function?